packages/components/src/components/SquiggleEditorImportedBindings.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { SquiggleEditor } from "./SquiggleEditor";
+import type { SquiggleEditorProps } from "./SquiggleEditor";
+import { runPartial, defaultBindings } from "@quri/squiggle-lang";
+import type { result, errorValue, bindings } from "@quri/squiggle-lang";
+
+function resultDefault(x: result<bindings, errorValue>): bindings {
+  switch (x.tag) {
+    case "Ok":
+      return x.value;
+    case "Error":
+      return defaultBindings;
+  }
+}
+
+function replaceBindings(
+  props: SquiggleEditorProps,
+  newBindings: bindings
+): SquiggleEditorProps {
+  return { ...props, bindings: newBindings };
+}
+
+export type SquiggleEditorImportedBindingsProps = SquiggleEditorProps & {
+  bindingsImportFile: string;
+};
+
+export const SquiggleEditorImportedBindings: React.FC<
+  SquiggleEditorImportedBindingsProps
+> = (props) => {
+  const [bindingsResult, setBindingsResult] = React.useState({
+    tag: "Ok",
+    value: defaultBindings,
+  } as result<bindings, errorValue>);
+  React.useEffect(() => {
+    async function retrieveBindings(fileName: string) {
+      let contents = await fetch(fileName).then((response) => {
+        return response.text();
+      });
+      setBindingsResult(runPartial(contents));
+    }
+    retrieveBindings(props.bindingsImportFile);
+  }, []);
+  const deliveredBindings = resultDefault(bindingsResult);
```

Comment:
I think this should just fail fast with `ErrorAlert` if execution failed. And also handle fetch errors by rendering and error.

For example, your current issue with `z is not defined` is due to the wrong url (it requests https://deploy-preview-892--squiggle-documentation.netlify.app/docs/squiggle/demo.squiggle instead of https://deploy-preview-892--squiggle-documentation.netlify.app/squiggle/demo.squiggle and gets 404).

Reply:
Oh, there's also the "loading" state to consider. Ideally, you should set the boolean `loading` state initially and not render the editor until the file is fetched.

Reply:
> is due to the wrong url

That doesn't explain why it works sometimes on netlify, and works all the time locally.

Reply:
Uh, no idea. It never works for me on netlify. Maybe docusaurus copies the static files to `docs/` when running locally, for some reason? But that would be surprising.

Reply:
@Hazelfire any ideas why it works sometimes on netlify and all the time locally?

Reply:
I figured it out, it depends on the trailing slash. Netlify always adds it, local dev server doesn't.

Reply:
thanks!

Reply:
![2022-07-28_10-25](https://user-images.githubusercontent.com/39039420/181545249-29eda9c5-3c04-44a2-8420-4154030aecfb.png)

we're allegedly already disabling asset optimization tho?

Reply:
well, `curl` doesn't care
```
qd@fw ~/D/P/B/Q/s/p/components (mdx-cli)> curl https://deploy-preview-892--squiggle-documentation.netlify.app/squiggle/demo.squiggle
x = 1 to 2
y = {a: x, b: 1e1}
f(t) = normal(t, 1.1)
z = y.b * y.a
qd@fw ~/D/P/B/Q/s/p/components (mdx-cli)> curl https://deploy-preview-892--squiggle-documentation.netlify.app/squiggle/demo.squiggle/
x = 1 to 2
y = {a: x, b: 1e1}
f(t) = normal(t, 1.1)
z = y.b * y.a
```

Reply:
Not the `.squiggle` url, but the webpage url in your browser. Your `../../squiggle/demo.squiggle` path is relative, so its resolution depends on whether there's a trailing slash in your current url. The easiest way to fix is to make the path absolute, `/squiggle/demo.squiggle`.